docs/user_manual.md

```diff
@@ -49,40 +60,277 @@ python -m virtualenv venv
+  the package. It requires installation from sources (using `--no-binary`):
+  ```bash
+  DATUMARO_HEADLESS=1 pip install datumaro --no-binary=datumaro
+  ```
+  This option can't be covered by extras due to Python packaging system
+  limitations.
+
+- Although Datumaro excludes `pycocotools` of version 2.0.2 in
+  requirements, it works with this version perfectly fine. The
+  reason for such requirement is binary incompatibility of the `numpy`
+  dependency in the `TensorFlow` and `pycocotools` binary packages,
+  and the current workaround forces this package to be build from sources
+  on most platforms
+  (see [#253](https://github.com/openvinotoolkit/datumaro/issues/253)).
+  If you need to use 2.0.2, make sure it is linked with the same version
+  of `numpy` as `TensorFlow` by reinstalling the package:
+  ``` bash
+  pip uninstall pycocotools
+  pip install pycocotools --no-binary=pycocotools
+  ```
+
+- When installing directly from the repository, you can change the
+  installation branch with `...@<branch_name>`. Also use `--force-reinstall`
+  parameter in this case. It can be useful for testing of unreleased
+  versions from GitHub pull requests.
+
+## How to use Datumaro
+
+As a standalone tool or a Python module:
 
 ``` bash
 datum --help
-```
-
-As a python module:
-> The directory containing Datumaro should be in the `PYTHONPATH`
-> environment variable or `cvat/datumaro/` should be the current directory.
 
-``` bash
 python -m datumaro --help
 python datumaro/ --help
 python datum.py --help
 ```
 
-As a python library:
+As a Python library:
 
 ``` python
-import datumaro
+from datumaro.components.project import Project
+from datumaro.components.dataset import Dataset
+from datumaro.components.extractor import Label, Bbox, DatasetItem
+...
+dataset = Dataset.import_from(path, format)
+...
+```
+
+### Glossary
+
+- Basic concepts:
+  - dataset - A number of media (dataset items) and associated annotations
+  - project - A combination of multiple datasets, plugins, models and metadata
+
+- Project versioning concepts:
+  - data source - a link to a dataset or a copy of a dataset inside a project.
+    Basically, an URL + dataset format name
+  - project revision - a commit hash or a named reference from
+    Git (branch, tag, HEAD~3 etc.).
+  - working / head / revision tree - a project build tree and plugins at
+    a specified revision
+  - data source revision - a data source hash at a specific stage
+  - object - a tree or a data source revision data
+
+- Dataset path concepts: <a id="revpath"></a>
+  - source / dataset / revision / project path - a path to a dataset in a
+    special format
+
+    - (project local) **rev**ision **path**s - a way to specify the path
+      to a source revision in the CLI, the syntax is:
+      `<revision>:<source/target name>`, any part can be omitted.
+      - Default revision is the working tree of the project
+      - Default target is the compiled project
```

Comment:
What is a "compiled" project? This word is only mentioned twice in the document and not explained.